data/sea/54-zebra.h

```diff
@@ -68,44 +88,72 @@ static ierror_msg_t zebra_translate
     zebra_attribute_t *attribute = &src->attributes[attribute_ix];
     iint_t columns_count = attribute->table.column_count;
 
+    /* zebra columns must match the number of input variables */
     if (columns_count != columns_to_fill) {
-        return ierror_msg_format ("zebra_translate: attribute_ix=%d, column_count=%d, columns_to_fill=%d\n", attribute_ix, columns_count, columns_to_fill);
+        return ierror_msg_format
+          ( "zebra_translate: attribute_ix=%d, column_count=%d, columns_to_fill=%d\n"
+          , attribute_ix
+          , columns_count
+          , columns_to_fill );
      }
 
     zebra_table_t *table = &attribute->table;
     iint_t row_count = table->row_count;
     iint_t fact_count = 0;
 
-    /* temporary fix until icicle uses the same time as zebra */
-    itime_t zebra_chord_time = itime_to_epoch_seconds (chord_time);
-
-    for (iint_t i = 0; i != row_count; ++i) {
-        itime_t time = attribute->times[i];
-        if (zebra_chord_time <= time) {
-            break;
+    printf ("fact_offset %lld, entity_fact_count %lld, row_count %lld\n", state->entity_fact_offset, state->entity_fact_count, row_count);
+    /* number of total facts for this entity */
+    if (state->entity_fact_count == 0) {
+        /* temporary fix until icicle uses the same time as zebra */
+        itime_t zebra_chord_time = itime_to_epoch_seconds (chord_time);
+
+        /* the number of facts before chord time for this entity */
+        for (iint_t i = 0; i != row_count; ++i) {
+            itime_t time = attribute->times[i];
+            if (zebra_chord_time <= time) {
+                break;
+            }
+            fact_count++;
         }
-        fact_count++;
+        state->entity_fact_count = fact_count;
+    } else {
+        fact_count = state->entity_fact_count;
     }
 
+    /* number of unread facts for this entity */
+    iint_t fact_offset    = state->entity_fact_offset;
+    iint_t fact_remaining = fact_count - fact_offset;
+
+    if (fact_remaining > zebra_chunk_fact_count) {
+        fact_count = zebra_chunk_fact_count;
+        state->entity_fact_offset += fact_count;
+    } else {
+        fact_count = fact_remaining;
```

Comment:
as in maybe it could get a fresh name at this point, rather than re-using the `fact_count` you were incrementing above, like `fact_chunk_count` or something? /not sure